src/main/scala/com/fulcrumgenomics/bam/CollectErccMetrics.scala

```diff
@@ -0,0 +1,253 @@
+          val concentration = erccData(name)
+          ErccDetailedMetric(name=name, concentration=concentration, count=count)
+        }.toSeq
+      Metric.write[ErccDetailedMetric](detailedPath, metrics)
+
+      // And try plotting
+      if (metrics.isEmpty || totalReads == 0 || simpleRegression.getN <= 1) {
+        logger.warning("No metrics generated (is your BAM empty?). Plots will not be generated.")
+      }
+      else {
+        val name = PathUtil.basename(input, trimExt=true).toString
+        Rscript.execIfAvailable(ScriptPath, detailedPath.toString, plotPath.toString, name, minTranscriptCount.toString) match {
+          case Failure(e) => logger.warning(s"Generation of PDF plots failed: ${e.getMessage}")
+          case _ => Unit
+        }
+      }
+    }
+  }
+
+  /** The natural log of 2! */
+  private val lnOf2 = scala.math.log(2) // natural log of 2
+
+  /** Computes the log base 2. */
+  private def log2(x: Double): Double = scala.math.log(x) / lnOf2
+
+  /** Use this method to wrap the calculation of correlations such that if insufficient data are given, zero is returned. */
+  private def zeroIfNaNOrInsufficientData(f: => Double): Double = {
+    try {
+      zeroIfNaN(f)
+    } catch {
+      case e: MathIllegalArgumentException => 0
+    }
+  }
+
+  /** Converts all `NaN` values to zero. */
+  private def zeroIfNaN(d: Double): Double = {
+    if (d.isNaN) 0 else d
+  }
+}
+
+/**
+  * Metrics produced by `CollectErccMetrics` describing various per-transcript metrics related to the spike-in of ERCC
+  * (External RNA Controls Consortium) into an RNA-Seq experiment.  One metric per ERCC transcript will be present.
+  *
+  * @param name the name (or ID) of the ERCC transcript.
+  * @param concentration the expected concentration as input to `CollectErccMetrics`.
+  * @param count the observed count of the # of fragments.
+  */
+case class ErccDetailedMetric
+(
+  name: String,
+  concentration: Double,
+  count: Long
+) extends Metric
+
+/**
+  * Metrics produced by `CollectErccMetrics` describing various summary metrics related to the spike-in of ERCC
+  * (External RNA Controls Consortium) into an RNA-Seq experiment.
+  *
+  * The correlation coefficients and linear regression are calculated based on the log2 observed fragment count versus
+  * the log2 expected concentration.
+  *
+  * @param total_reads the total # of reads considered.
+  * @param ercc_reads the total # of reads mapping to an ERCC transcript.
+  * @param pct_ercc_reads the fraction of total reads that map to an ERCC transcript.
+  * @param ercc_fragments the total # of read pairs mapping to an ERCC transcript.
+  * @param total_transcripts the total # of ERCC transcripts with at least one read observed.
+  * @param total_filtered_transcripts the total # of ERCC transcripts with at least the user-set minimum # of reads observed.
+  * @param pearsons_correlation the Pearson's correlation coefficient.
```

Comment:
I may be wrong, but it's weird to me to have "the" before "Pearson's correlation coefficient" and likewise for Spearman's.